### DISCUSSION TITLE
Custom theme color does not appear in VS Code IntelliSense suggestions

### DISCUSSION THREAD

**Discussion by rumtraubenuss** [chakra-ui/chakra-ui#3226] [Q&A] [answered]
I have a typescript next.js project using `@chakra-ui/react@1.2.1`. I am extending the basic theme with chakra's `extendTheme` function to add a custom color. The custom color is rendered when I use the custom color identifier as a `color` prop on a chakra component. What I am missing is that this custom color is suggested by IntelliSense in VS Code when I do a ctrl-space on the `color` prop on chakra component. All the basic theme colors are listed but not the custom one. Is this a general behavior or am I missing something? 

Edit:

Find a corresponding codesandbox here: https://codesandbox.io/s/happy-microservice-r6m8d?file=/src/index.tsx
The custom color `anotherColor` renders the `Heading` red. If you alter the `color` prop to only contain an `a` and press ctrl-space you will get code completion suggestions. `anotherColor` is suggested but have a look at the icon: the `abc` means that this is only suggested because it is a string found in the current file and not from the type definition like the other colors marked with icon with two horizontal lines. If you would use this in a file where the extended theme is not defined you would get no `anotherColor` suggestion at all.

![screen_codesandbox](https://user-images.githubusercontent.com/1382052/106395658-c3f19100-6403-11eb-9626-19ef3d99566d.png)

BTW: I sometimes "managed" to make the code completion not to work at all in codesandbox. Reloading the example several times made it work again. No idea what this is related to.

**Comment by with-heart** [comment]
Hi @rumtraubenuss! Could you create a CodeSandbox of your setup? It's hard to say what's going wrong here without being able to see your theme.

**Reply by rumtraubenuss** [reply]
@with-heart Thanks for your feedback. I updated the question with a codesandbox example.

**Comment by dodas** [comment, marked as answer]
Hi @rumtraubenuss,

support for custom theme typing has been added only recently.
You will need to generate custom typings using Chakra UI CLI.
[Read the `Theme Typings` docs here](https://chakra-ui.com/docs/theming/advanced#theme-typings).

**Reply by rumtraubenuss** [reply]
Thanks @dodas for pointing out!
I missed this. RTFM 😬

**Reply by jeromeSH26** [reply]
Hi, I'm using this cli as explained in the doc, but the intellisense is not showing the custom colors..

```
import { extendTheme } from "@chakra-ui/react";

const themeExtension = {
  colors: {
      primary: "blue",
  },
};

const theme = extendTheme(themeExtension);

export default theme;
```

then run the cli using npx

`npx chakra-cli tokens ./styles/customTheme/theme.ts
`

can't see the primary color
![image](https://user-images.githubusercontent.com/51024915/126140194-26184fd4-6ae6-4abb-a814-dd3d70290af4.png)

**Reply by jarodburchill** [reply]
> Hi, I'm using this cli as explained in the doc, but the intellisense is not showing the custom colors..
> 
> ```
> import { extendTheme } from "@chakra-ui/react";
> 
> const themeExtension = {
>   colors: {
>       primary: "blue",
>   },
> };
> 
> const theme = extendTheme(themeExtension);
> 
> export default theme;
> ```
> 
> then run the cli using npx
> 
> `npx chakra-cli tokens ./styles/customTheme/theme.ts `
> 
> can't see the primary color
> ![image](https://user-images.githubusercontent.com/51024915/126140194-26184fd4-6ae6-4abb-a814-dd3d70290af4.png)

Me too :/

**Reply by tannguyencse19** [reply]
`chakra-ui@1.7.3`

This problem has been fixed.
